components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -5732,18 +5732,25 @@ def _getQueryString(cls, opts=None):
         Extend base query to handle filtering of Datasets by Projects.
         Returns a tuple of (query, clauses, params).
         Supported opts: 'project': <project_id> to filter by Project
+                        'image': <image_id> to filter by child Image
                         'orphaned': <bool>. Filter by 'not in Project'
 
         :param opts:        Dictionary of optional parameters.
         :return:            Tuple of string, list, ParametersI
         """
         query, clauses, params = super(
             _DatasetWrapper, cls)._getQueryString(opts)
-        if opts is not None and 'project' in opts:
+        if opts is None:
+            opts = {}
+        if 'project' in opts:
             query += ' join obj.projectLinks plink'
```

Comment:
minor: to align with the rest of the query, you could use ``projectlinks`` instead of ``plink``